src/browser/BrowserService.cpp

```diff
@@ -1149,9 +1149,14 @@ QJsonObject BrowserService::prepareEntry(const Entry* entry)
 {
     QJsonObject res;
 #ifdef WITH_XC_BROWSER_PASSKEYS
-    // Use Passkey's username instead if found
-    res["login"] = entry->hasPasskey() ? passkeyUtils()->getUsernameFromEntry(entry)
-                                       : entry->resolveMultiplePlaceholders(entry->username());
+    // If entry has a passkey, but the passkey username differs from the normal username, return the normal username
+    auto username = entry->resolveMultiplePlaceholders(entry->username());
+    auto passkeyUsername = passkeyUtils()->getUsernameFromEntry(entry);
+    if (entry->hasPasskey()) {
+        res["login"] = username != passkeyUsername ? username : passkeyUsername;
```

Comment:
This ternary operator doesn't make sense. If they are equal than returning either username or passkeyUsername would result in the same outcome. If they are unequal then you return the username. So just return the username.

Looking at the code change in general, you always just return the username.

I think you need to write a test case against this behavior to validate the desired outcome.

Reply:
In other words, you need to better define when you actually want to return the passkeyUsername and use that as your branching condition.

Reply:
Hmm. I guess you are right. It should be enough to just always return the username. `KPEX_PASSKEY_USERNAME` is used for the authentication anyway, and the original reason we added the username in a passkey entry in a first place was just for filtering.